### ISSUE TITLE
W8003 doesn't recognise aws:PrincipalAccount as a 

### ISSUE THREAD

**Issue by waddamski** [aws-cloudformation/cfn-lint#2956]
### CloudFormation Lint Version

0.82.2

### What operating system are you using?

Ubuntu

### Describe the bug

When validating the output of an AWS CloudFormation Template, cfn-lint errors as follows:
`W8003 Fn::Equals element will alway return false`
This is for the following template section:
```
WSTemplateFormatVersion: '2010-09-09'
Description: Account Users
Conditions:
  is-webops-development-env:
    Fn::Equals: [ "aws:PrincipalAccount", "12345674891"]
```
It almost seems to be that cfn-lint recognises "aws:PrincipalAccount" as a string rather than a [global condition context key](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html#condition-keys-principalaccount) and so rightly suggests that the condition will never be true.

### Expected behavior

The expected behaviour is that the possibility of interpolation is allowed and the condition can be true.

### Reproduction template

Create an AWS Cloudformation template with the following:
```
AWSTemplateFormatVersion: '2010-09-09'
Description: Account Users
Conditions:
  is-webops-development-env:
    Fn::Equals: [ "aws:PrincipalAccount", "12345674891"]
```
Or as follows:
```
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Conditions": {
    "is-permitted-account": {
      "Fn::Equals": [
        "aws:PrincipalAccount",
        "123456789"
      ]
    }
  }
}
```
run cfn-lint against it.

**Comment by kddejong** [comment]
`aws:PrincipalAccount` is a context key for IAM policies and not for CloudFormation.

I think what you are looking for is
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: Account Users
Conditions:
  is-webops-development-env:
    Fn::Equals: [ !Ref "AWS::AccountId", "12345674891"]
```

Of course you could be looking to compare a different account ID from the account you are working in in which case a different approach would have to be used.

**Comment by waddamski** [comment]
Many thanks for the swift response!
Yes you're right - we've refactored. 
Thanks again!
